src/main/java/com/tabgok/entity/Machine.java

```diff
@@ -0,0 +1,54 @@
+package com.tabgok.entity;
+
+import com.tabgok.data.StringData;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * The machine is a model of a machine.  Importantly, the machine is the top-level
+ * model for a machine, meaning it contains within it either references to, or values of,
+ * everything within a machine from Hardware to Filesystems to applications.
+ */
+public class Machine extends Entity {
+    private StringData MachineID;
+    private final HashSet<String> mountedFilesystems = new HashSet<>();
+    
+    private Machine(){}
+    
+    public static Machine getMachine(){
+        return new Machine();
+    }
+    
+    /**
+     * 
+     * @return Returns the machine's ID - this ID is expected to be globally unique
+     */
+    public StringData getID(){
+        return MachineID;
+    }
+    
+    /**
+     * 
+     * @return Return the active mounted partition points on the machine
+     */
+    public String[] getmountedFilesystemIDs(){
```

Comment:
there are really good arguments for always passing around immutable data structures, and very few good arguments for passing around mutable ones. This is a good example of why mutable data structures are problematic -- you have to make a copy of it to return it from a method, and (something you missed here) you have to worry about concurrency issues.

at google, for example, this would return an ImmutableSet, and mountedFilesystems would be an ImmutableSet, so we could just return it.

Two options I can think of here:
 - Find a library with immutable data structures that you like and use that (the one we use is known publicly as guava)
 - just return a Set\<String\> that is a copy of the member variable; nobody wants an array.